app/interactors/book_suggestions_by_genre.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class BookSuggestionsByGenre
+  include Interactor
+
+  def call
+    genre = Book.find(context.book_id).genre
+    books_by_genre = Book.filter(genre: genre).to_a
```

Comment:
This will instantiate all of the records filtered, which can be a lot, and might crash the server!
You can use `find_each` will work on a collection in batches.

Or how about filtering books whose name is not included in the names of books rented (using something like `user.rents.pluck(:name).uniq`)?